### PR TITLE
[GH-10035] Added the cypress spec for cancel out leave a team modal

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
Added the cypress test for the case "cancel out leave a team modal"

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/10035

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: cypress/integration/team/teams_spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+describe('Teams Suite', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('Cancel out of leaving a team', () => {
+        // * check the team name
+        cy.get('#headerTeamName').should('contain', 'eligendi');
```
Comment (saturninoabril) [inline cypress/integration/team/teams_spec.js]: Also please add initial check to team URL, like:
```javascript
cy.url().should('include', '/ad-1/channels/town-square');
```

File: cypress/integration/team/teams_spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        // 2. open the drop down menu
+        cy.get('#sidebarHeaderDropdownButton').should('be.visible').click();
+
+        // 3. click the leave team
+        cy.get('#sidebarDropdownMenu #leaveTeam').should('be.visible').click();
+
```
Comment (saturninoabril) [inline cypress/integration/team/teams_spec.js]: Please add element ID to leave team modal (e.g. `leaveTeamModal` [here](https://github.com/mattermost/mattermost-webapp/blob/eef36a3149f37cb59762f8fbb023e097aa3e0905/components/leave_team_modal/leave_team_modal.jsx#L80)) so we could add validation that the leave team modal has opened up after clicking above, like:
```javascript
// * Check that the "leave team modal" opened up
cy.get('#leaveTeamModal').should('be.visible');
```

File: cypress/integration/team/teams_spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        // 4. click on no
+        cy.get('.modal-content .modal-footer .btn-default').should('be.visible');
+        cy.get('.modal-content .modal-footer .btn-default').should('contain', 'No');
+        cy.get('.modal-content .modal-footer .btn-default').click();
```
Comment (saturninoabril) [inline cypress/integration/team/teams_spec.js]: Above selector is fragile and might break in the future.  Instead, you may add ID to "No" button like `leaveTeamNo` [here](https://github.com/mattermost/mattermost-webapp/blob/eef36a3149f37cb59762f8fbb023e097aa3e0905/components/leave_team_modal/leave_team_modal.jsx#L100).

File: cypress/integration/team/teams_spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (saturninoabril) [inline cypress/integration/team/teams_spec.js]: Please add validation that the  `leaveTeamModal` has closed after clicking "No" above.

File: cypress/integration/team/teams_spec.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        // * check the team name
+        cy.get('#headerTeamName').should('contain', 'eligendi');
```
Comment (saturninoabril) [inline cypress/integration/team/teams_spec.js]: Finally, please add check to team URL that it's still the same.

**Comment by pradeepmurugesan** [comment]
@saturninoabril made the changes. Kindly review

**Pull request merged**